plugins/i3s/io/EsriReader.cpp

```diff
@@ -112,42 +111,83 @@ namespace pdal
         {
             dimData data;
             std::string readName = attributes[i]["name"].asString();
-            int key = std::stoi(attributes[i]["key"].asString());
-            data.key = key;
+            data.name = readName;
+            data.key = std::stoi(attributes[i]["key"].asString());
+            data.dataType =
+                attributes[i]["attributeValues"]["valueType"].asString();
+
 
             if(attributes[i].isMember("valueType"))
             {
-                std::string type = attributes[i]["valueType"].asString();
-                data.dataType = type;
+                data.dataType = attributes[i]["valueType"].asString();
             }
 
-            else if (readName == "RGB")
+
+            if (readName == "RGB")
             {
                 layout->registerDim(Dimension::Id::Red);
                 layout->registerDim(Dimension::Id::Green);
                 layout->registerDim(Dimension::Id::Blue);
                 // Since RGB are always packed together and handled specially,
                 // we'll use Red as our indicator that RGB exists.
+
                 m_dimMap[Dimension::Id::Red] = data;
             }
+            //Available dimension types can be found at https://git.io/fAbxS
             else if (m_dimensions.find(readName) != m_dimensions.end())
             {
-                layout->registerDim(m_dimensions.at(readName));
-                m_dimMap[m_dimensions.at(readName)] = data;
-
+                data.dimType = Dimension::Type::None;
+                if(data.dataType == "UInt8")
```

Comment:
IMO this kind of thing belongs in a function.  Perhaps just in dimType?
```
using namespace Dimension;
std::map<std::string, Dimension::Type> dimmap { {"Uint8", Type::Unsigned8}, {"Uint16", Type::Unsigned16}, ...

auto di = dimmap.find(name);
return (di != dimmap.end() ? di->second : Type::Unknown);
```
To me, this is just cleaner and doesn't clutter up your code.  You can find() in a vector, too, but in practical terms, it makes no difference.